modules/Microsoft.Web/hostingEnvironments/deploy.bicep

```diff
@@ -139,28 +188,53 @@ resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (ena
   }
 }
 
-resource appServiceEnvironment 'Microsoft.Web/hostingEnvironments@2021-03-01' = {
+resource appServiceEnvironment 'Microsoft.Web/hostingEnvironments@2022-03-01' = {
   name: name
   kind: kind
   location: location
   tags: tags
+  identity: identity
   properties: {
+    clusterSettings: clusterSettings
+    dedicatedHostCount: dedicatedHostCount != 0 ? dedicatedHostCount : null
+    dnsSuffix: dnsSuffix
+    frontEndScaleFactor: frontEndScaleFactor
+    internalLoadBalancingMode: internalLoadBalancingMode
+    ipsslAddressCount: ipsslAddressCount != 0 ? ipsslAddressCount : null
+    multiSize: !empty(multiSize) ? any(multiSize) : null
+    upgradePreference: upgradePreference
+    userWhitelistedIpRanges: !empty(userWhitelistedIpRanges) ? userWhitelistedIpRanges : null
     virtualNetwork: {
       id: subnetResourceId
       subnet: last(split(subnetResourceId, '/'))
     }
-    internalLoadBalancingMode: internalLoadBalancingMode
-    multiSize: !empty(multiSize) ? any(multiSize) : null
-    ipsslAddressCount: ipsslAddressCount != -1 ? ipsslAddressCount : null
-    dnsSuffix: dnsSuffix
-    frontEndScaleFactor: frontEndScaleFactor
-    clusterSettings: clusterSettings
-    userWhitelistedIpRanges: !empty(userWhitelistedIpRanges) ? userWhitelistedIpRanges : null
-    dedicatedHostCount: dedicatedHostCount != -1 ? dedicatedHostCount : null
     zoneRedundant: zoneRedundant
   }
 }
 
+module appServiceEnvironment_configurations_networking 'configurations-networking/deploy.bicep' = if (kind == 'ASEv3') {
```

Comment:
What is the reason for 2 children instead of the API-aligned common `configurations` child?

I know in case of the AppService we have a similar implementation - but there the reason were the vastly different properties depending on the type. Is this a similar situation here?

Reply:
the properties are different based on the name (networking or customdnssuffix). If they are common, I would have followed a similar approach to vm extensions

Reply:
ok, too bad. I guess we could have also used a `union` to just apply the relevant properties. However, the user would still be exposed to all parameters (as parameter sets are not a thing) which I guess is a hassle. I just hope there'll never be a child-resource type to this. :D